src/core/page-parser.ts

```diff
@@ -1,67 +1,96 @@
+        const lodestonePage = await this.getLodestonePage(characterId).catch(error => {
+            if (axios.isAxiosError(error) && error.response?.status) {
+                throw new Error(String(error.response.status));
+            }
+            throw error;
         });
-        const dom = new JSDOM(data);
-        let {document} = dom.window;
-        const columnsQuery = req.query['columns'];
+
+        const dom = new JSDOM(lodestonePage);
         const selectors = this.getCSSSelectors();
-        let columns: string[];
-        if (columnsQuery && !Array.isArray(columnsQuery)) {
-            columns = [columnsQuery.toString()]
-                .filter(column => {
-                    return column.startsWith(columnsPrefix)
-                })
-                .map(column => column.replace(columnsPrefix, ''));
-        } else if (columnsQuery && Array.isArray(columnsQuery)) {
-            columns = columnsQuery.map(c => c.toString())
-                .filter(column => {
-                    return column.startsWith(columnsPrefix)
-                })
-                .map(column => column.replace(columnsPrefix, ''));
-        } else {
-            columns = Object.keys(selectors).map(key => {
-                return this.definitionNameToColumnName(key);
-            }).filter(column => column !== 'default');
-        }
-        return columns.reduce((acc, column) => {
+        const columnsToParse = columns.length > 0 ? columns : Object.keys(selectors).map(this.definitionNameToColumnName).filter(column => column !== 'default');
+        
+        let {document} = dom.window;
+        let data: Record<string, unknown> = {};
+
+        for (const column of columnsToParse) {
             const definition = this.getDefinition(selectors, column);
+            
             if (column === 'Root') {
                 const context = this.handleColumn(definition, document)?.data;
                 const contextDOM = new JSDOM(context);
                 document = contextDOM.window.document;
-                return {
-                    ...acc
-                }
-            }
-            const parsed = this.handleColumn(definition, document);
-            if (parsed.isPatch || column === 'Entry') {
-                return {
-                    ...acc,
-                    ...(parsed.data || {})
+            } else {
+                const parsed = this.handleColumn(definition, document);
+                
+                if (parsed.isPatch || column === 'Entry') {
+                    data = {
+                        ...data,
+                        ...(parsed.data || {}),
+                    };
+                } else {
+                    data[column] = parsed.data;
                 }
             }
-            return {
-                ...acc,
-                [column]: parsed.data
-            }
-        }, {});
+        }
+
+        return this.modifyParseResult(data);
```

Comment:
What would `modifyParseResult` be used for? Adding game data like what XIVAPI does? Should probably name it `extendParseResult` imo, `modify` is too broad and could easily lead to "I'm doing a lot of shit" implementations.

Also consider making it async if you want it to use game data, since it might have to query XIVAPI itself to grab that data and then cache it, making it possibly async.

Reply:
After thinking about it a bit more, I think we should drop this helper function completely.

I used it to get the mount and minion parsers more consistent with the XIVAPI character endpoints "MIMO" feature, since just using the current lodestone-css-selectors results in a nested object like this:
```js
Mount: {
  Mount: {
    List: [ ... ]
  },
  Total: 20
}
```

That specifically should probably by solved in the lodestone-css-selectors, but if other subclasses of `PageParser` want to modify the parsing output, they should just overwrite the `get` method and call super.get() in it anyway.